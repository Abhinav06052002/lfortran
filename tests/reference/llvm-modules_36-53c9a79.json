{
    "basename": "llvm-modules_36-53c9a79",
    "cmd": "lfortran --no-color --show-llvm {infile} -o {outfile}",
    "infile": "tests/../integration_tests/modules_36.f90",
    "infile_hash": "81bd7e21e30c46c3779cc4c928723fbf58bcc748333a96faee9d0b96",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "llvm-modules_36-53c9a79.stdout",
    "stdout_hash": "9d25f3a8c0f768c66bc85f2e19106d7f4323e67c0f969a85707e382b",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}