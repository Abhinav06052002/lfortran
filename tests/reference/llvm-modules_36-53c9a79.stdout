; ModuleID = 'LFortran'
source_filename = "LFortran"

%fpm_run_settings_polymorphic = type { i64, %fpm_run_settings* }
%fpm_run_settings = type { %fpm_build_settings, [2 x i8*], i8*, i8*, i1 }
%fpm_build_settings = type { i1 }
%__vtab_fpm_run_settings = type { i64 }

@0 = private unnamed_addr constant [1 x i8] zeroinitializer, align 1

define i1 @Any_4_1_0_logical____0(i1* %mask, i32* %__1mask, i32* %__2mask) {
.entry:
  %Any_4_1_0 = alloca i1, align 1
  %__1_i = alloca i32, align 4
  store i1 false, i1* %Any_4_1_0, align 1
  %0 = load i32, i32* %__1mask, align 4
  %1 = sub i32 %0, 1
  store i32 %1, i32* %__1_i, align 4
  br label %loop.head

loop.head:                                        ; preds = %loop.body, %.entry
  %2 = load i32, i32* %__1_i, align 4
  %3 = add i32 %2, 1
  %4 = load i32, i32* %__2mask, align 4
  %5 = load i32, i32* %__1mask, align 4
  %6 = add i32 %4, %5
  %7 = sub i32 %6, 1
  %8 = icmp sle i32 %3, %7
  br i1 %8, label %loop.body, label %loop.end

loop.body:                                        ; preds = %loop.head
  %9 = load i32, i32* %__1_i, align 4
  %10 = add i32 %9, 1
  store i32 %10, i32* %__1_i, align 4
  %11 = load i1, i1* %Any_4_1_0, align 1
  %12 = load i32, i32* %__1_i, align 4
  %13 = load i32, i32* %__1mask, align 4
  %14 = load i32, i32* %__2mask, align 4
  %15 = sub i32 %12, %13
  %16 = mul i32 1, %15
  %17 = add i32 0, %16
  %18 = mul i32 1, %14
  %19 = getelementptr inbounds i1, i1* %mask, i32 %17
  %20 = load i1, i1* %19, align 1
  %21 = icmp eq i1 %11, false
  %22 = select i1 %21, i1 %20, i1 %11
  store i1 %22, i1* %Any_4_1_0, align 1
  br label %loop.head

loop.end:                                         ; preds = %loop.head
  br label %return

return:                                           ; preds = %loop.end
  %23 = load i1, i1* %Any_4_1_0, align 1
  ret i1 %23
}

define void @__module_modules_36_fpm_main_01_cmd_run(%fpm_run_settings_polymorphic* %settings, i1* %test) {
.entry:
  %call_arg_value1 = alloca i32, align 4
  %call_arg_value = alloca i32, align 4
  %0 = alloca %__vtab_fpm_run_settings, align 8
  %1 = getelementptr %__vtab_fpm_run_settings, %__vtab_fpm_run_settings* %0, i32 0, i32 0
  store i64 0, i64* %1, align 4
  %__libasr_created__intrinsic_array_function_Any = alloca [2 x i1], align 1
  %__libasr_created__intrinsic_array_function_Any1 = alloca i1, align 1
  %__libasr_index_0_ = alloca i32, align 4
  %__libasr_index_0_1 = alloca i32, align 4
  %found = alloca [2 x i1], align 1
  %toomany = alloca i1, align 1
  store i32 1, i32* %__libasr_index_0_1, align 4
  store i32 0, i32* %__libasr_index_0_, align 4
  br label %loop.head

loop.head:                                        ; preds = %loop.body, %.entry
  %2 = load i32, i32* %__libasr_index_0_, align 4
  %3 = add i32 %2, 1
  %4 = icmp sle i32 %3, 2
  br i1 %4, label %loop.body, label %loop.end

loop.body:                                        ; preds = %loop.head
  %5 = load i32, i32* %__libasr_index_0_, align 4
  %6 = add i32 %5, 1
  store i32 %6, i32* %__libasr_index_0_, align 4
  %7 = load i32, i32* %__libasr_index_0_, align 4
  %8 = sub i32 %7, 1
  %9 = mul i32 1, %8
  %10 = add i32 0, %9
  %11 = getelementptr [2 x i1], [2 x i1]* %__libasr_created__intrinsic_array_function_Any, i32 0, i32 %10
  %12 = load i32, i32* %__libasr_index_0_1, align 4
  %13 = sub i32 %12, 1
  %14 = mul i32 1, %13
  %15 = add i32 0, %14
  %16 = getelementptr [2 x i1], [2 x i1]* %found, i32 0, i32 %15
  %17 = load i1, i1* %16, align 1
  %18 = xor i1 %17, true
  store i1 %18, i1* %11, align 1
  %19 = load i32, i32* %__libasr_index_0_1, align 4
  %20 = add i32 %19, 1
  store i32 %20, i32* %__libasr_index_0_1, align 4
  br label %loop.head

loop.end:                                         ; preds = %loop.head
  %21 = getelementptr [2 x i1], [2 x i1]* %__libasr_created__intrinsic_array_function_Any, i32 0, i32 0
  store i32 1, i32* %call_arg_value, align 4
  store i32 2, i32* %call_arg_value1, align 4
  %22 = call i1 @Any_4_1_0_logical____0(i1* %21, i32* %call_arg_value, i32* %call_arg_value1)
  store i1 %22, i1* %__libasr_created__intrinsic_array_function_Any1, align 1
  %23 = load i1, i1* %__libasr_created__intrinsic_array_function_Any1, align 1
  %24 = load i1, i1* %toomany, align 1
  %25 = load i1, i1* %test, align 1
  %26 = xor i1 %25, true
  %27 = icmp eq i1 %24, false
  %28 = select i1 %27, i1 %24, i1 %26
  %29 = load i1, i1* %toomany, align 1
  %30 = getelementptr %fpm_run_settings_polymorphic, %fpm_run_settings_polymorphic* %settings, i32 0, i32 1
  %31 = load %fpm_run_settings*, %fpm_run_settings** %30, align 8
  %32 = getelementptr %fpm_run_settings, %fpm_run_settings* %31, i32 0, i32 3
  %33 = load i8*, i8** %32, align 8
  %34 = alloca i8*, align 8
  store i8* %33, i8** %34, align 8
  %35 = alloca i8*, align 8
  store i8* getelementptr inbounds ([1 x i8], [1 x i8]* @0, i32 0, i32 0), i8** %35, align 8
  %36 = call i1 @_lpython_str_compare_noteq(i8** %34, i8** %35)
  %37 = icmp eq i1 %29, false
  %38 = select i1 %37, i1 %29, i1 %36
  %39 = icmp eq i1 %28, false
  %40 = select i1 %39, i1 %38, i1 %28
  %41 = getelementptr %fpm_run_settings_polymorphic, %fpm_run_settings_polymorphic* %settings, i32 0, i32 1
  %42 = load %fpm_run_settings*, %fpm_run_settings** %41, align 8
  %43 = getelementptr %fpm_run_settings, %fpm_run_settings* %42, i32 0, i32 0
  %44 = getelementptr %fpm_build_settings, %fpm_build_settings* %43, i32 0, i32 0
  %45 = load i1, i1* %44, align 1
  %46 = xor i1 %45, true
  %47 = icmp eq i1 %40, false
  %48 = select i1 %47, i1 %40, i1 %46
  %49 = icmp eq i1 %23, false
  %50 = select i1 %49, i1 %48, i1 %23
  br i1 %50, label %then, label %else

then:                                             ; preds = %loop.end
  br label %ifcont

else:                                             ; preds = %loop.end
  br label %ifcont

ifcont:                                           ; preds = %else, %then
  br label %return

return:                                           ; preds = %ifcont
  ret void
}

declare i1 @_lpython_str_compare_noteq(i8**, i8**)

define i32 @main(i32 %0, i8** %1) {
.entry:
  %call_arg_value = alloca i1, align 1
  call void @_lpython_call_initial_functions(i32 %0, i8** %1)
  %settings = alloca %fpm_run_settings, align 8
  %2 = getelementptr %fpm_run_settings, %fpm_run_settings* %settings, i32 0, i32 2
  %3 = getelementptr %fpm_run_settings, %fpm_run_settings* %settings, i32 0, i32 4
  %4 = getelementptr %fpm_run_settings, %fpm_run_settings* %settings, i32 0, i32 1
  %5 = getelementptr %fpm_run_settings, %fpm_run_settings* %settings, i32 0, i32 3
  %6 = alloca %fpm_run_settings_polymorphic, align 8
  %7 = getelementptr %fpm_run_settings_polymorphic, %fpm_run_settings_polymorphic* %6, i32 0, i32 0
  store i64 0, i64* %7, align 4
  %8 = getelementptr %fpm_run_settings_polymorphic, %fpm_run_settings_polymorphic* %6, i32 0, i32 1
  store %fpm_run_settings* %settings, %fpm_run_settings** %8, align 8
  store i1 true, i1* %call_arg_value, align 1
  call void @__module_modules_36_fpm_main_01_cmd_run(%fpm_run_settings_polymorphic* %6, i1* %call_arg_value)
  call void @_lpython_free_argv()
  br label %return

return:                                           ; preds = %.entry
  ret i32 0
}

declare void @_lpython_call_initial_functions(i32, i8**)

declare void @_lpython_free_argv()
